### PR TITLE
VMware: correct documentation for datacenter

### LINKED ISSUE #38290: vmware_guest_find has required, deprecated parameter
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

vmware_guest_find

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = None
  configured module search path = [u'/home/jhriv/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jhriv/ansible/local/lib/python2.7/site-packages/ansible
  executable location = /home/jhriv/ansible/bin/ansible
  python version = 2.7.13 (default, Nov 24 2017, 17:33:09) [GCC 6.3.0 20170516]
```

##### OS / ENVIRONMENT

Distributor ID:	Debian
Description:	Debian GNU/Linux 9.4 (stretch)
Release:	9.4
Codename:	stretch

##### SUMMARY

`vmware_gest_find` with datacenter: parameter:
>DEPRECATION WARNING]: Param 'datacenter' is deprecated. See the module docs for more information. This feature will
be removed in version 2.9. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.

`vmware_gest_find` without datacenter: parameter:
> fatal: [localhost]: FAILED! => {"changed": false, "msg": "missing required arguments: datacenter"}

##### STEPS TO REPRODUCE

Follow the example on http://docs.ansible.com/ansible/latest/modules/vmware_guest_find_module.html

The "datacenter:" parameter is missing

```yaml
# From upstream documentation
- name: Find Guest's Folder using name
  vmware_guest_find:
    hostname: 192.168.1.209
    username: administrator@vsphere.local
    password: vmware
    validate_certs: no
    name: testvm
  register: vm_folder

- name: Find Guest's Folder using UUID
  vmware_guest_find:
    hostname: 192.168.1.209
    username: administrator@vsphere.local
    password: vmware
    validate_certs: no
    uuid: 38c4c89c-b3d7-4ae6-ae4e-43c5118eae49
  register: vm_folder
```

##### EXPECTED RESULTS

```
 _______________________________________
< TASK [Find Guest's Folder using name] >
 ---------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
```
##### ACTUAL RESULTS

Error

```
fatal: [localhost]: FAILED! => {"changed": false, "msg": "missing required arguments: datacenter"}
```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This fixes documentation related to datacenter in vmware_guest_find
module.

Fixes: #38290

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_guest_find.py

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py [hunk 1]
```diff
@@ -28,22 +28,22 @@
     - PyVmomi
 options:
    name:
-        description:
-            - Name of the VM to work with.
-            - This is required if uuid is not supplied.
+     description:
+     - Name of the VM to work with.
+     - This is required if C(uuid) parameter is not supplied.
```
Comment (gundalow) [inline lib/ansible/modules/cloud/vmware/vmware_guest_find.py:33]: Do we also want `mutually_exclusive=(['name', 'uuid'],),`?

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py [hunk 2]
```diff
@@ -28,22 +28,22 @@
...
+     - This is required if C(name) parameter is not supplied.
    datacenter:
-        description:
-            - Destination datacenter for the find operation.
-            - Deprecated in 2.5, will be removed in 2.9 release.
-        required: True
+     description:
+     - Destination datacenter for the find operation.
+     - Deprecated in 2.5, will be removed in 2.9 release.
+     required: False
```
Comment (gundalow) [inline lib/ansible/modules/cloud/vmware/vmware_guest_find.py]: You can removed `required: false` lines as that's the default

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py [hunk 3]
```diff
@@ -112,7 +111,7 @@ def main():
     argument_spec.update(
         name=dict(type='str'),
         uuid=dict(type='str'),
-        datacenter=dict(removed_in_version=2.9, type='str', required=True)
+        datacenter=dict(removed_in_version=2.9, type='str', required=False)
```
Comment (gundalow) [inline lib/ansible/modules/cloud/vmware/vmware_guest_find.py]: You can remove `, required=False`

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_guest_find.py:33]: Done.

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_guest_find.py]: Done.

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py (hunk 3 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_guest_find.py]: Done.

**Review by gundalow — APPROVED** [review]
Looks good. Should be cherry-picked into `stable-2.5`

**Comment by Akasurde** [comment]
@gundalow Thanks for review. I will cherry-pick for stable-2.5

**Pull request merged**